File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/logic_ternary.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeLogicTernary(statement: JSONObject, onError: (e: Exception) -> Unit): Any? {
    try {
        Log.d("hello", "execute_logic_ternary")
        val pointer = statement.getJSONObject("pointer")
        val cond = executeStatement(pointer["cond"].toString()) ?: false
        if (cond as Boolean) {
            return executeStatement(pointer["value1"].toString())
        }
        return executeStatement(pointer["value2"].toString())
    } catch (e: Exception) {
        return onError(e)
    }
}

File: piper/extensions/chrome/src/engine/statements/logic_ternary.js
import { stringify } from "../util"


export default async function(statement, onError) {
    try {
        let pointer = statement.pointer
        let cond = await this.executeStatement(stringify(pointer.cond)) || false
        if (cond) {
            return await this.executeStatement(stringify(pointer.value1))
        }
        return await this.executeStatement(stringify(pointer.value2))
    } catch (e) {
        return onError(e)
    }
}
